### ISSUE TITLE
Checking if player has item that was set to variable always return false result

### ISSUE THREAD

**Issue by MorkaZ** [SkriptLang/Skript#2045]
### Description
```
command testtt:
	trigger:
		set {_tool.in.variable} to player's tool # Setting player's tool to variable
		if {_tool.in.variable} is player's tool: # Just make sure that skript sees that value of this variable is equal to player's tool
			broadcast "variable is player tool"
			if player has {_tool.in.variable}: # Now the problem - this condition always will return false probably because it checks for item that is same object, not if it is equal.
				broadcast "It works!" # Dreaming about it..
			else:
				broadcast "it does not works." # This message always appear
			remove 1 of {_tool.in.variable} from player # And here skripts detects that play has this item and it takes an item from inventory.
			add 1 of {_tool.in.variable} to player's inventory
		loop items of inventory of player:
			if loop-value is {_tool.in.variable}:
				broadcast "found! %loop-value%" # And here is example how skript will everytime find this item just by looping inventory. Please, fix it..
				stop
```

### Steps to Reproduce
Set item to variable and use:
```
if player has {_item.in.variable}:
```

### Expected Behavior
Return true if given item is equal to one of items in inventory.

### Errors / Screenshots
0 errors.

### Server Information
* **Server version/platform:**   2 weeks ago builded 1.13.2 spigot server  
* **Skript version:**   2.3.6

### Additional Context
Addons: MorkazSk, SkQuery by LimeGlass, skRayFall+v1.9.14, TuSKe 1.8.3-PikachuPatch-v3, skDragon2dev

**Comment by TheBentoBox** [comment]
I can confirm this though it's kinda strange. I did the following tests while holding 1 bonemeal:
* `if player has player's tool` returned true
* `set {test} to player's tool` and `if player has {test}` returned false (this is incorrect)
* `set {test} to 1 bonemeal` and `if player has {test}` returned true

Definitely an issue here though, maybe when setting a variable to a slot?

**Comment by Moderocky** [comment]
There has always been some issue with variables and slots, though I thought this was fixed in 2.3.6. The old issue was that it set the variable to an 'instance' of the player's tool, so if it changed then the variable would no longer be the same.

I suppose the way to test would be to put the variable from your second test into a string `"%{test}%" and see what it was actually spitting out.

The old trick to fix this was to use `random item out of {test}` or similar, as this forced it to return an item-stack. Perhaps this could be a temporary solution here?

**Comment by TheBentoBox** [comment]
This is fixed in 2.4-alpha2.
